crates/router/src/core/customers.rs

```diff
@@ -614,143 +612,394 @@ pub async fn delete_customer(
     Ok(services::ApplicationResponse::Json(response))
 }
 
-#[cfg(all(
-    any(feature = "v1", feature = "v2", feature = "oltp"),
-    not(feature = "customer_v2")
-))]
 #[instrument(skip(state))]
 pub async fn update_customer(
     state: SessionState,
     merchant_account: domain::MerchantAccount,
-    update_customer: customers::CustomerRequest,
+    update_customer: customers::CustomerUpdateRequest,
     key_store: domain::MerchantKeyStore,
+    id: Option<String>,
```

Comment:
Since in V1, flow handler won't have Global_id, sent to it by the v1 merchant

Reply:
Because having same handler, don't handle as option.
- instead have two sepatate function or
- create a type which is feature flagged and have two version of get function, since the return type is different use a inline cfg. (Instead using inline cfg's the v2 function can return Result also)